components/brave_wallet/browser/bitcoin/bitcoin_transaction.cc

```diff
@@ -223,6 +224,13 @@ BitcoinTransaction::TxInput::FromValue(const base::Value::Dict& value) {
     return std::nullopt;
   }
 
+  // Enforce a conservative upper bound for witness stack serialization to
+  // reject obviously invalid or abusive inputs. Standard script element limit
+  // is 520 bytes; we apply the same ceiling to the serialized witness.
+  if (result.witness.size() > 520) {
+    return std::nullopt;
+  }
+
   if (!ReadOptionalHexByteArrayTo(value, "raw_outpoint_tx",
```

Comment:
This is our internal format of persisting transactions in db. We either add simliar reach validation to everything in a consistent manner or just stay with simple validations we have now(string is a string, etc)